### PR TITLE
[Issue-212] Upgrade to python 3.10.15

### LINKED ISSUE #212: Upgrade Python
We are currently on 3.9. 

We should be on 3.10, at least. Upgrading to 3.10 will facilitate upgrading to Django 5.0: https://docs.djangoproject.com/en/5.1/releases/5.0/

### PR DESCRIPTION
**Pull request by reginafcompton**
This PR upgrades the AHS site from python 3.9 to 3.10.15. Closes #212 

I approached the upgrade like so:

1. I read release notes for relevant breaking changes: https://docs.python.org/3/whatsnew/3.10.html I did not find anything relevant.
2. I changed the python version in `Pipfile`; I recreated my virtual env (i.e., with `pipenv install -d`....which also update the `Pipfile.lock`).
3. I changed other references to python 3.9: Dockerfile and github workflow config.

Then, I ran unit tests:

```
pipenv run pytest
```

And perhaps most importantly, I rebuilt the docker image and QA'd the site manually:
```
docker compose up
```

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,4 +1,3 @@
-version: "3"
```
Comment (reginafcompton) [inline docker-compose.yml:1]: This change has nothing to do with Python.

...but the "version" tag is obsolete, and I was tired of seeing the warning from Docker.

**Comment by reginafcompton** [comment]
@ecedmondson - do you have time to look at this PR? ensure that I did not overlook something critical or trivial or silly?

File: Dockerfile [hunk 2]
```diff
@@ -10,6 +10,7 @@ RUN apt-get install --no-install-recommends --assume-yes curl
 
 COPY ./Pipfile /app/Pipfile
 COPY ./Pipfile.lock /app/Pipfile.lock
+RUN pip install --upgrade pip
```
Comment (reginafcompton) [inline Dockerfile:13]: https://github.com/americanhandelsociety/americanhandelsociety-members/issues/212#issuecomment-2489351886

Comment (reginafcompton) [reply Dockerfile:13]: This warning has been present for some time. This PR seems like an acceptable place to fix it.

**Comment by ecedmondson** [comment]
Of course! It will probably be tomorrow or Wednesday. I hope that's alright @reginafcompton .

**Comment by reginafcompton** [comment]
thank you @ecedmondson ! sounds great.

**Review by ecedmondson — APPROVED** [review]
I tested the main flows (except join other organizations, because I realized I don't know anything about that) and everything worked great!

I often find this useful, so I thought I would share: https://www.nicholashairs.com/posts/major-changes-between-python-versions/

Happy thanksgiving!

**Pull request merged**